crates/weaver_resolved_schema/src/lib.rs

```diff
@@ -92,23 +146,375 @@ impl ResolvedTelemetrySchema {
+        let mut changes = SchemaChanges::new();
+
+        if let Some(ref manifest) = self.registry_manifest {
+            changes.set_head_manifest(weaver_version::schema_changes::RegistryManifest {
+                semconv_version: manifest.semconv_version.clone(),
+            });
+        }
+
+        if let Some(ref manifest) = baseline_schema.registry_manifest {
+            changes.set_baseline_manifest(weaver_version::schema_changes::RegistryManifest {
+                semconv_version: manifest.semconv_version.clone(),
+            });
+        }
+
+        // Attributes in the registry
+        self.diff_attributes(baseline_schema, &mut changes);
+
+        // Signals
+        let latest_signals = self.groups(GroupType::Metric);
+        let baseline_signals = baseline_schema.groups(GroupType::Metric);
+        self.diff_signals(
+            SchemaItemType::Metrics,
+            &latest_signals,
+            &baseline_signals,
+            &mut changes,
+        );
+        let latest_signals = self.groups(GroupType::Event);
+        let baseline_signals = baseline_schema.groups(GroupType::Event);
+        self.diff_signals(
+            SchemaItemType::Events,
+            &latest_signals,
+            &baseline_signals,
+            &mut changes,
+        );
+        let latest_signals = self.groups(GroupType::Span);
+        let baseline_signals = baseline_schema.groups(GroupType::Span);
+        self.diff_signals(
+            SchemaItemType::Spans,
+            &latest_signals,
+            &baseline_signals,
+            &mut changes,
+        );
+        let latest_signals = self.groups(GroupType::Resource);
+        let baseline_signals = baseline_schema.groups(GroupType::Resource);
+        self.diff_signals(
+            SchemaItemType::Resources,
+            &latest_signals,
+            &baseline_signals,
+            &mut changes,
+        );
+
+        changes
+    }
+
+    fn diff_attributes(
+        &self,
+        baseline_schema: &ResolvedTelemetrySchema,
+        changes: &mut SchemaChanges,
+    ) {
+        let latest_attributes = self.registry_attribute_map();
+        let baseline_attributes = baseline_schema.registry_attribute_map();
+
+        // A map of attributes that have been renamed to a new attribute.
+        // The key is the new name of the attribute, and the value is a set of old names.
+        // The key may refer to an existing attribute in the baseline schema or
+        // a new attribute in the latest schema.
+        let mut renamed_attributes = HashMap::new();
+
+        // ToDo for future PR, process differences at the field level (not required for the schema update)
+
+        // Collect all the information related to the attributes that have been
+        // deprecated in the latest schema.
+        for (attr_name, attr) in latest_attributes.iter() {
+            if let Some(deprecated) = attr.deprecated.as_ref() {
```

Comment:
I think there needs to be a comparison with the baseline here, otherwise all deprecated attributes will be flagged as changes.
```suggestion
            if let Some(deprecated) = attr.deprecated.as_ref() {
                // is this a change from the baseline?
                if let Some(baseline_attr) = baseline_attributes.get(attr_name) {
                    if let Some(baseline_deprecated) = baseline_attr.deprecated.as_ref() {
                        if deprecated == baseline_deprecated {
                            continue;
                        }
                    }
                }
```